src/main/java/apoc/periodic/Periodic.java

```diff
@@ -395,18 +422,28 @@ public Pattern regNoCaseMultiLine(String pattern) {
                     }
                 }
             }*/
-            collector.incrementCount(currentBatchSize);
-        } while (iterator.hasNext());
-
-        boolean wasTerminated = Util.transactionIsTerminated(terminationGuard);
-        ToLongFunction<Future<Long>> toLongFunction = wasTerminated ?
-                f -> Util.getFutureOrCancel(f, collector.getBatchErrors(), collector.getFailedBatches(), 0L) :
-                f -> Util.getFuture(f, collector.getBatchErrors(), collector.getFailedBatches(), 0L);
-        collector.incrementSuccesses(futures.stream().mapToLong(toLongFunction).sum());
-
-        Util.logErrors("Error during iterate.commit:", collector.getBatchErrors(), log);
-        Util.logErrors("Error during iterate.execute:", collector.getOperationErrors(), log);
-        return Stream.of(collector.getResult());
+                collector.incrementCount(currentBatchSize);
+            } while (iterator.hasNext());
+
+            boolean wasTerminated = Util.transactionIsTerminated(terminationGuard);
+            ToLongFunction<Future<Long>> toLongFunction = wasTerminated ?
+                    f -> Util.getFutureOrCancel(f, collector.getBatchErrors(), collector.getFailedBatches(), 0L) :
+                    f -> Util.getFuture(f, collector.getBatchErrors(), collector.getFailedBatches(), 0L);
+            collector.incrementSuccesses(futures.stream().mapToLong(toLongFunction).sum());
+
+            Util.logErrors("Error during iterate.commit:", collector.getBatchErrors(), log);
+            Util.logErrors("Error during iterate.execute:", collector.getOperationErrors(), log);
+            return Stream.of(collector.getResult());
+        } finally {
+            if (parallel==false) { // we need to ensure destruction of the executor we've created above
+                pool.shutdown();
+                try {
+                    pool.awaitTermination(10, TimeUnit.SECONDS);
```

Comment:
what happens if the update statements are long running statements ...
either give it much more time like 10 or 30 mins
or ... configurable? or ...

the single pool was meant to avoid this creating and shutting down pools but serialize everything